tactics/equality.ml

```diff
@@ -988,7 +984,7 @@ let discrimination_pf env sigma e (t,t1,t2) discriminator lbeq =
   let sigma, i           = build_coq_I sigma in
   let sigma, absurd_term = build_coq_False sigma in
   let eq_elim, eff       = ind_scheme_of_eq lbeq in
-  let sigma, eq_elim = Evd.fresh_global (Global.env ()) sigma eq_elim in
+  let sigma, eq_elim = Evd.fresh_global env sigma eq_elim in
```

Comment:
Here it is wrong, the whole discrimination_pf should be put in the monad, and ind_scheme_of_eq as well, propagating the effect directly using tclEFFECTS (it can generate the scheme on the fly, leading to your Not_found exception here, as the scheme is not present in env).